solr/core/src/test/org/apache/solr/schema/TestCloudManagedSchema.java

```diff
@@ -53,21 +54,22 @@ public void test() throws Exception {
     QueryRequest request = new QueryRequest(params);
     request.setPath("/admin/cores");
     int which = r.nextInt(clients.size());
-    HttpSolrClient client = (HttpSolrClient) clients.get(which);
-    String previousBaseURL = client.getBaseURL();
-    // Strip /collection1 step from baseURL - requests fail otherwise
-    client.setBaseURL(previousBaseURL.substring(0, previousBaseURL.lastIndexOf("/")));
-    NamedList<?> namedListResponse = client.request(request);
-    client.setBaseURL(previousBaseURL); // Restore baseURL
-    NamedList<?> status = (NamedList<?>) namedListResponse.get("status");
-    NamedList<?> collectionStatus = (NamedList<?>) status.getVal(0);
-    String collectionSchema = (String) collectionStatus.get(CoreAdminParams.SCHEMA);
-    // Make sure the upgrade to managed schema happened
-    assertEquals(
-        "Schema resource name differs from expected name", "managed-schema.xml", collectionSchema);
 
-    SolrZkClient zkClient = new SolrZkClient(zkServer.getZkHost(), 30000);
-    try {
+    // create a client that does not have the /collection1 as part of the URL.
+    try (SolrClient rootClient =
+        new HttpSolrClient.Builder(buildUrl(jettys.get(which).getLocalPort())).build()) {
+      NamedList<?> namedListResponse = rootClient.request(request);
+      NamedList<?> status = (NamedList<?>) namedListResponse.get("status");
```

Comment:
There's a multi argument get or something on named list that lets you drill in. Don't remember what it's called exactly, I'm on my phone now.

Reply:
I'd love a suggestion on this, or even a quick commit, as this whole area is a bit of dark art to me!